### PR TITLE
add v2 blog for dns-migration

### PR DESCRIPTION
**Pull request by premdeepsaini**

### CODE REVIEW

File: content/blog/is-dns-migration-a-tricky-affair.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+Before starting, if you are here because of the catchy title and have no idea of what DNS is, or you are one of my many (zero) followers or need to brush up your knowledge on DNS, please refer to these resources:
+1. [What is DNS - Cloudflare](https://www.cloudflare.com/learning/dns/what-is-dns/)
+2. [Deep Dive on DNS - IETF](https://www.youtube.com/watch?v=DV0q9s94RL8)
+
+I won't keep you waiting any longer - I will start my story and explain the reason for my introductory lines.
+# Need for DNS migration
+One of our clients wanted us to move their workload from docker swarm to Amazon ECS. Earlier, they were using HaProxy to load-balance the traffic. This setup also included generation and installation of SSL certificates using in-house scripts, which the client needed to maintain. So, we decided to go with a managed solution to reduce the management overhead and chose AWS ALB and ACM combination. In order for ACM to auto-renew expiring SSL certificates, we decided to use Route53 as the DNS service, ditching GoDaddy's DNS service. BTW, if you are curious, the domain was registered with GoDaddy and hence their DNS service.
```
Comment (vjdhama) [inline content/blog/is-dns-migration-a-tricky-affair.md:23]: s/docker swarm/Docker Swam/g

Comment (vjdhama) [reply content/blog/is-dns-migration-a-tricky-affair.md:23]: Maybe we can reword. 

s/So, we decided to go with a managed solution to reduce the management overhead and chose AWS ALB and ACM combination/Since we were moving to ECS, ALB and ACM felt like an obvious choice/g

File: content/blog/is-dns-migration-a-tricky-affair.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+Still, as a good engineer 🤭, I went ahead and did my research regarding DNS migration and carved out a migration plan.  Looking at the plan for migration, even my instincts were convinced. 
+
+
+![Instinct calling](/images/blog/is-dns-migration-a-ticky-affair/do_it.gif)
+# Migration Plan
+Basically I divided the migration plan in three phases along with a rollback strategy:
+## Pre-execution phase
+In this phase we exported the [zone file](https://en.wikipedia.org/wiki/Zone_file) from GoDaddy DNS service which we will be referring to for the list of domain records to migrate.
+
+Another thing we did was to reduce the [TTL](https://www.cloudflare.com/learning/cdn/glossary/time-to-live-ttl/) of all the records in GoDaddy DNS to the least possible value. These tasks we performed a day before the execution phase. For us, the maximum value of TTL was one day, so 24 hours was enough for us to reflect new TTL values.
```
Comment (vjdhama) [inline content/blog/is-dns-migration-a-tricky-affair.md:36]: Maybe rephrase the last line. Sounds little weird, tbh.

File: content/blog/is-dns-migration-a-tricky-affair.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+## Execution Phase
+This phase included changes at Route53's end. This included:
+1. Addition of hosted zone in Route53 for the [apex domain](https://help.easyredir.com/en/articles/453072-what-is-a-domain-apex)
```
Comment (vjdhama) [inline content/blog/is-dns-migration-a-tricky-affair.md:40]: Can we find better reference for apex/root domain than that?

File: content/blog/is-dns-migration-a-tricky-affair.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+2. Addition of all the records we have in the zone file to Route53 hosted zone we just created.
+3. Update the nameserver records provided by Route53 in the domain registration configuration. This will switch the DNS service to Route53 once the [TTL](https://www.cloudflare.com/learning/cdn/glossary/time-to-live-ttl/) is expired.
+We actually used Terraform to add the hosted zone as well as other records, as this would ease up the maintenance of the records.
+
+## Verification Phase
+This phase included verification of the application and mail server. One way to verify this is by simply heading to the application domain and sending a mail to the colleague. This can be considered as a correct way to verify since we have taken care of the TTL, although, to make sure the DNS query result is being served by Route53 nameservers we have to make use of a linux utility called -
```
Comment (vjdhama) [inline content/blog/is-dns-migration-a-tricky-affair.md:46]: See if this reword makes sense?

s/This phase included verification of the application and mail server/This phase included verification of DNS records of our application and mail servers/g

Comment (vjdhama) [reply content/blog/is-dns-migration-a-tricky-affair.md:46]: Break before *although*? Can start another sentence.

File: content/blog/is-dns-migration-a-tricky-affair.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+![me crying](/images/blog/is-dns-migration-a-ticky-affair/bug.gif)
+
+
+Others in my team were trying to console me by showing tweets of failed DNS migrations and how common this is. Just kidding!! They were like
+![panicing](/images/blog/is-dns-migration-a-ticky-affair/panic.gif)
+
+Well, this is a bit exaggerated. 
+# Troubleshooting
+With the verification using dig, we were sure that migration did happen and DNS had migrated to Route53. This eliminated the chance of issue in nameserver configuration at the domain registrar end. 
+Next we moved to Route53 to check the domain records for the email server and application. It was at this point we noticed something unusual.
```
Comment (vjdhama) [inline content/blog/is-dns-migration-a-tricky-affair.md:71]: Maybe insert lens gif? 😄

File: content/blog/is-dns-migration-a-tricky-affair.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+When we copied the records from the zone file exported from GoDaddy DNS service, we copied the name of the records as it is. In a lot of DNS services "@" is used to represent the apex domain i.e. **my_cool_domain.com**. We assumed it to be the same for Route53 as well and applied the changes. This led to creation of subdomain @.my_cool_domain.com in Route53.
```
Comment (vjdhama) [inline content/blog/is-dns-migration-a-tricky-affair.md:73]: Invalid domain 😆 

s/my_cool_domain.com/my-cool-domain.in/g

File: content/blog/is-dns-migration-a-tricky-affair.md (hunk 1 above)
Comment (premdeepsaini) [reply content/blog/is-dns-migration-a-tricky-affair.md:23]: Fixed in 15acecd

File: content/blog/is-dns-migration-a-tricky-affair.md (hunk 2 above)
Comment (premdeepsaini) [reply content/blog/is-dns-migration-a-tricky-affair.md:36]: Fixed in 15acecd

File: content/blog/is-dns-migration-a-tricky-affair.md (hunk 3 above)
Comment (premdeepsaini) [reply content/blog/is-dns-migration-a-tricky-affair.md:40]: Fixed in 15acecd

File: content/blog/is-dns-migration-a-tricky-affair.md (hunk 4 above)
Comment (premdeepsaini) [reply content/blog/is-dns-migration-a-tricky-affair.md:46]: Fixed in 15acecd

File: content/blog/is-dns-migration-a-tricky-affair.md (hunk 5 above)
Comment (premdeepsaini) [reply content/blog/is-dns-migration-a-tricky-affair.md:71]: fixed in 15acecd

File: content/blog/is-dns-migration-a-tricky-affair.md (hunk 6 above)
Comment (premdeepsaini) [reply content/blog/is-dns-migration-a-tricky-affair.md:73]: Fixed in 15acecd

**Comment by vjdhama** [comment]
LGTM 🚀
